bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/WebexTeamsHandler.java

```diff
@@ -0,0 +1,315 @@
+
+    private @Nullable ScheduledFuture<?> refreshService = null;
+
+    public WebexTeamsHandler(Thing thing, OAuthFactory oAuthFactory, HttpClient httpClient) {
+        super(thing);
+        this.oAuthFactory = oAuthFactory;
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No commands supported on any channel
+    }
+
+    @Override
+    public void handleRemoval() {
+        updateStatus(ThingStatus.REMOVED);
+    }
+
+    // creates list of available Actions
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singletonList(WebexTeamsActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing...");
+        config = getConfigAs(WebexTeamsConfiguration.class);
+
+        updateStatus(this.status);
+
+        final String token = config.token;
+        final String clientId = config.clientId;
+        final String clientSecret = config.clientSecret;
+
+        if (!token.isBlank() && clientId.isBlank()) { // For bots
+            logger.debug("I'm a bot");
+        } else if (!clientId.isBlank()) { // For integrations
+            logger.debug("I'm a person.");
+            if (clientSecret.isEmpty()) {
+                this.status = ThingStatus.OFFLINE;
+                updateStatus(this.status, ThingStatusDetail.CONFIGURATION_ERROR, "@text/confErrorNoSecret");
+                return;
+            }
+        } else {
+            this.status = ThingStatus.OFFLINE;
+            updateStatus(this.status, ThingStatusDetail.CONFIGURATION_ERROR, "@text/confErrorTokenOrId");
+            return;
+        }
+
+        // background initialization:
+        scheduler.execute(() -> {
+            try {
+                this.client = new WebexTeamsApi(this, oAuthFactory, httpClient);
+                logger.debug("Trying to fetch account details");
+                Person p = this.client.getPerson();
+                logger.debug("Success: {}", p.getDisplayName());
+
+                this.status = ThingStatus.ONLINE;
+                updateStatus(this.status);
+
+                refresh();
+
+            } catch (NotAuthenticatedException e) {
+                logger.error("Failed to initialize client", e);
+                this.status = ThingStatus.OFFLINE;
+                updateStatus(this.status, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+        });
+
+        // TODO: make interval configurable.
+        refreshService = scheduler.scheduleWithFixedDelay(this::refresh, 0, 300, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing...");
+        final ScheduledFuture<?> job = refreshService;
+        if (job != null) {
+            job.cancel(true);
+            refreshService = null;
+        }
+        this.client.dispose();
```

Comment:
Please check that client is not null. It will be null in case of configuration error for example.

Reply:
Code has been removed as part of refactoring.